src/utils/StreamHighlightUtils.ts

```diff
@@ -82,13 +82,22 @@ export function getRestHighlightsForAllTerms(
   opts: IStreamHighlightOptions
 ): IHighlight[] {
   const indexes = [];
-  const uniqueTermsToHighlight = getUniqueTermsToHighlight(termsToHighlight, phrasesToHighlight);
+  const termsFromPhrases = _.chain(phrasesToHighlight)
+    .values()
+    .map(_.keys)
+    .flatten()
+    .value();
 
-  _.each(uniqueTermsToHighlight, (term: string) => {
-    let termsToIterate = _.compact([term].concat(termsToHighlight[term]).sort(termsSorting));
-    termsToIterate = _.map(termsToIterate, term => Utils.escapeRegexCharacter(term));
-    let regex = regexStart;
-    regex += termsToIterate.join('|') + ')(?=(?:' + wordBoundary + '|$)+)';
+  _.each(termsToHighlight, (terms: string[], term: string) => {
+    const termsToIterate = _.chain([term])
```

Comment:
The name `termsToIterate` is a little generic. Wouldn't these be the `termsToHighlight`?

Reply:
It would be confusing to call them termsToHighlight as it is already a variable sent by the API that we use to iterate in the first place... 🤔

Reply:
What about `uniqueTermsToHighlight` since the result is after a diff against `termsFromPhrases`?

Reply:
Perfect 👍